### ISSUE TITLE
Asset loading depends on the "multi-threaded" feature being enabled

### ISSUE THREAD

**Issue by komadori** [bevyengine/bevy#10209]
## Bevy version

5d110eb96e64eda03d3f3716d2e94555a68ec846 (0.12-dev)

## Relevant system information

```ignore
AdapterInfo { name: "NVIDIA GeForce 940MX", vendor: 4318, device: 4941, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "535.113.01", backend: Vulkan }
SystemInfo { os: "Linux 21.2 Linux Mint", kernel: "5.15.0-86-generic", cpu: "Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz", core_count: "4", memory: "31.2 GiB"
```

## What you did

Run the animated_fox example using a subset of Bevy's default features that was sufficient with the previous release (0.11).

```cargo run --release --example animated_fox --no-default-features --features="animation,bevy_asset,bevy_scene,bevy_winit,bevy_core_pipeline,bevy_pbr,bevy_gltf,bevy_render,ktx2,zstd,png,x11,tonemapping_luts"```

## What went wrong

The fox asset is not displayed at all, only the background.

However, if I add the "multi-threaded" feature then the example works correctly. Is it intentional that this feature is required for asset loading to work?

**Comment by alice-i-cecile** [comment]
I suspect this is a general problem of asset loading currently. This code should fail to compile if it doesn't run.

That said, I think I'd prefer to enable single-threaded asset loading.

**Comment by robojeb** [comment]
I spent some time on my lunch. and it seems like there is something racy going on between the `async-fs` IO executor and the single threaded local executor bevy is using. 

Async code isn't my forte so I am not entirely sure where the mismatch is, but if I set a breakpoint on `raw.rs:644` inside `async-task` (v4.5.0) and I step through slowly, eventually the asset loads. 
But if I step through quickly, I get an `AssetEvent::Added` event, but eventually the LocalExecutor has no more items in its queue and so no more progress is made on actually loading the asset. 
It always seems to happen when going through an `async-fs` future.
